[beta] Rollup backports

pietroalbini opened the pull request:
Rolled up:

* [beta] adjust deprecation date of mem::uninitialized #61160 

Cherry-picked:

* Fix ICE with inconsistent macro matchers #61046 
* Revert "Add implementations of last in terms of next_back on a bunch of DoubleEndedIterators." #61057 
* debuginfo: Revert to old/more verbose behavior for -Cdebuginfo=1 #61007 

r? @ghost

pietroalbini commented:
@bors r+ p=20

bors commented:
:pushpin: Commit 242de1cbc8ed092631b0c2931005e8236e7e9b82 has been approved by `pietroalbini`

<!-- @bors r=pietroalbini 242de1cbc8ed092631b0c2931005e8236e7e9b82 -->
<!-- homu: {"type":"Approved","sha":"242de1cbc8ed092631b0c2931005e8236e7e9b82","approver":"pietroalbini"} -->

pietroalbini commented:
@bors r-

A PR had wrong labels, I'll add it.

ehuss commented:
I don't think #61110 should be included without #61098.  I suspect it won't pass tests without it.

pietroalbini commented:
@ehuss that's a really good point.

Removed https://github.com/rust-lang/rust/pull/61110, added https://github.com/rust-lang/rust/pull/61007.

@bors r+ p=20

bors commented:
:pushpin: Commit 4f22a662cd9e7f6b8d13727461af69cac78aec35 has been approved by `pietroalbini`

<!-- @bors r=pietroalbini 4f22a662cd9e7f6b8d13727461af69cac78aec35 -->
<!-- homu: {"type":"Approved","sha":"4f22a662cd9e7f6b8d13727461af69cac78aec35","approver":"pietroalbini"} -->

bors commented:
:hourglass: Testing commit 4f22a662cd9e7f6b8d13727461af69cac78aec35 with merge c3ffeea353cef7cab8ed31a9f55de31e63a892ac...
<!-- homu: {"type":"BuildStarted","head_sha":"4f22a662cd9e7f6b8d13727461af69cac78aec35","merge_sha":"c3ffeea353cef7cab8ed31a9f55de31e63a892ac"} -->

bors commented:
:broken_heart: Test failed - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/builds/24876465)
<!-- homu: {"type":"BuildFailed","builder_url":"https://ci.appveyor.com/project/rust-lang/rust/builds/24876465","builder_name":"status-appveyor"} -->

ehuss commented:
I have filed https://github.com/rust-lang/rls/issues/1466 for the flakey rls error since I've personally seen it happen multiple times.

pietroalbini commented:
@bors retry

bors commented:
:hourglass: Testing commit 4f22a662cd9e7f6b8d13727461af69cac78aec35 with merge 70d3e4e53d0b1389050083735af5123778296fc2...
<!-- homu: {"type":"BuildStarted","head_sha":"4f22a662cd9e7f6b8d13727461af69cac78aec35","merge_sha":"70d3e4e53d0b1389050083735af5123778296fc2"} -->

bors commented:
:broken_heart: Test failed - [checks-travis](https://travis-ci.com/rust-lang/rust/builds/113571575)
<!-- homu: {"type":"BuildFailed","builder_url":"https://travis-ci.com/rust-lang/rust/builds/113571575","builder_name":"checks-travis"} -->

rust-highfive commented:
The job `arm-android` of your PR [failed on Travis](https://travis-ci.com/rust-lang/rust/jobs/203852669) ([raw log](https://api.travis-ci.com/v3/job/203852669/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[01:40:22] test string::test_str_clear ... ok
[01:40:22] test string::test_str_truncate ... ok
[01:40:22] test string::test_str_truncate_invalid_len ... ok
[01:40:22] test string::test_str_truncate_split_codepoint ... ok
[01:40:22] died due to signal 11
[01:40:22] 
[01:40:22] 
[01:40:22] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--target" "arm-linux-androideabi" "-j" "4" "--release" "--locked" "--color" "always" "--features" "panic-unwind backtrace" "--manifest-path" "/checkout/src/libstd/Cargo.toml" "-p" "alloc" "--"
[01:40:22] expected success, got: exit code: 3
---
travis_time:end:0623e032:start=1559132479719264653,finish=1559132479733880604,duration=14615951
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:000789b0
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb --batch -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:247157b5
travis_time:start:247157b5
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:0d84b2a4
$ dmesg | grep -i kill
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

pietroalbini commented:
@bors retry

Android signal 11.

bors commented:
:hourglass: Testing commit 4f22a662cd9e7f6b8d13727461af69cac78aec35 with merge 0ff088c9ee6a132daf256383e56004bee34fbaae...
<!-- homu: {"type":"BuildStarted","head_sha":"4f22a662cd9e7f6b8d13727461af69cac78aec35","merge_sha":"0ff088c9ee6a132daf256383e56004bee34fbaae"} -->

bors commented:
:sunny: Test successful - [checks-travis](https://travis-ci.com/rust-lang/rust/builds/113587391), [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/builds/24893660)
Approved by: pietroalbini
Pushing 0ff088c9ee6a132daf256383e56004bee34fbaae to beta...
<!-- homu: {"type":"BuildCompleted","approved_by":"pietroalbini","base_ref":"beta","builders":{"status-appveyor":"https://ci.appveyor.com/project/rust-lang/rust/builds/24893660","checks-travis":"https://travis-ci.com/rust-lang/rust/builds/113587391"},"merge_sha":"0ff088c9ee6a132daf256383e56004bee34fbaae"} -->
